internal/commands/app/diff.go

```diff
@@ -0,0 +1,138 @@
+package app
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/atlas"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/local"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/AlecAivazis/survey/v2"
+
+	"github.com/spf13/pflag"
+)
+
+type diffOutputs struct {
+	noDiffs bool
+}
+
+type diffInputs struct {
+	AppDirectory        string
+	IncludeDependencies bool
+	IncludeHosting      bool
+	cli.ProjectInputs
+}
+
+// CommandDiff is the `app diff` command
+type CommandDiff struct {
+	inputs      diffInputs
+	outputs     diffOutputs
+	atlasClient atlas.Client
+	realmClient realm.Client
+}
+
+// Flags is the command flags
+func (cmd *CommandDiff) Flags(fs *pflag.FlagSet) {
+	fs.StringVarP(&cmd.inputs.AppDirectory, flagDirectory, flagDirectoryShort, "", flagDirectoryUsage)
+	fs.BoolVarP(&cmd.inputs.IncludeDependencies, flagIncludeDependencies, flagIncludeDependenciesShort, false, flagIncludeDependenciesUsage)
+	fs.BoolVarP(&cmd.inputs.IncludeHosting, flagIncludeHosting, flagIncludeHostingShort, false, flagIncludeHostingUsage)
+
+	cmd.inputs.Flags(fs)
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDiff) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Setup is the command setup
+func (cmd *CommandDiff) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+// Handler is the command handler
+func (cmd *CommandDiff) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, err := local.LoadApp(cmd.inputs.AppDirectory)
+	if err != nil {
+		return err
+	}
+
+	apps, err := cmd.realmClient.FindApps(realm.AppFilter{
```

Comment:
this chunk of logic for selecting an app _should_ be all covered in `cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter()` (it handles fetching the apps, errors if none are found, returns the single result if present, or otherwise presents the user with a prompt to pick the app they are targeting)

(and `cli.ProjectInputs` has a function defined on it to provide the `realm.AppFilter`)